server/src/core/server/services/dsaReports/reports.ts

```diff
@@ -20,3 +46,178 @@ export async function createDSAReport(
+  input: AddDSAReportShareInput,
+  now = new Date()
+) {
+  const result = await createReportShare(mongo, tenant.id, input, now);
+  const { dsaReport } = result;
+  return dsaReport;
+}
+
+export interface DeleteDSAReportNoteInput {
+  id: string;
+  reportID: string;
+}
+
+/**
+ * deleteDSAReportNote deletes a note from the history of a DSAReport
+ *
+ * @param mongo is the mongo context.
+ * @param tenant is the filtering tenant for this operation.
+ * @param input is the input used for deleting the note
+ * @returns the DSAReport with the note deleted from is history.
+ */
+export async function deleteDSAReportNote(
+  mongo: MongoContext,
+  tenant: Tenant,
+  input: DeleteDSAReportNoteInput,
+  now = new Date()
+) {
+  const result = await deleteReportNote(mongo, tenant.id, input);
+  const { dsaReport } = result;
+  return dsaReport;
+}
+
+export interface ChangeDSAReportStatusInput {
+  userID: string;
+  status: GQLDSAReportStatus;
+  reportID: string;
+}
+
+/**
+ * changeDSAReportStatus changes the status of a DSAReport
+ * and also adds the status change to its history
+ *
+ * @param mongo is the mongo context.
+ * @param tenant is the filtering tenant for this operation.
+ * @param input is the input used for changing the status
+ * @returns the DSAReport with its new status
+ */
+export async function changeDSAReportStatus(
+  mongo: MongoContext,
+  tenant: Tenant,
+  input: ChangeDSAReportStatusInput,
+  now = new Date()
+) {
+  const result = await changeReportStatus(mongo, tenant.id, input, now);
+  const { dsaReport } = result;
+  return dsaReport;
+}
+
+export interface MakeDSAReportDecisionInput {
+  commentID: string;
+  commentRevisionID: string;
+  userID: string;
+  reportID: string;
+  legality: GQLDSAReportDecisionLegality;
+  legalGrounds?: string;
+  detailedExplanation?: string;
+}
+
+/**
+ * makeDSAReportDecison makes an illegal vs legal decision for a DSAReport
+ * and also adds the decision to its history
+ *
+ * @param mongo is the mongo context.
+ * @param tenant is the filtering tenant for this operation.
+ * @param input is the input used for making the decision
+ * @returns the DSAReport with its decision
+ */
+export async function makeDSAReportDecision(
+  mongo: MongoContext,
+  redis: AugmentedRedis,
+  cache: DataCache,
+  config: Config,
+  broker: CoralEventPublisherBroker,
+  tenant: Tenant,
+  input: MakeDSAReportDecisionInput,
+  now = new Date()
+) {
+  // TODO: Send through rejection legalGrounds and detailedExplanation once backend support is available
+  const {
+    commentID,
+    commentRevisionID,
+    userID,
+    // legalGrounds,
+    // detailedExplanation,
+  } = input;
+
+  // REJECT if ILLEGAL
+  if (input.legality === GQLDSAReportDecisionLegality.ILLEGAL) {
+    await rejectComment(
+      mongo,
+      redis,
+      cache,
+      config,
+      broker,
+      tenant,
+      commentID,
+      commentRevisionID,
+      userID,
+      now
+    );
+    // APPROVE IF NOT ILLEGAL
```

Comment:
I'm unsure about approving a comment that is determined not illegal. Even if it's not illegal, it could still be toxic? Are we sure we don't want to just leave it with `NONE` and remain visible? Approving is like we're rubber stamping it from our mod team as good.

Reply:
Good point, removing as agree that we don't want to do that! Thanks for catching.